### PR TITLE
FileSearcher

### PR DESCRIPTION
**Pull request by johanhummel**

### CODE REVIEW

**Review by carlofthetower — APPROVED** [review]
LGTM

File: FileSearcher.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+public class FileSearcher {
+    
+    public static void searchFile(String fileName, String pattern)  throws IOException {
+        try (BufferedReader br = new BufferedReader(new FileReader(fileName))) {
+
+            String line;
+
+            while ((line = br.readLine()) != null) {
+                if (line.contains(pattern)) {
```
Comment (AlexanderSandelius) [inline FileSearcher.java:13]: Should it be case-sensitive?

File: FileSearcher.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        if (args.length != 3 || !"search".equals(args[0])) {
+            System.out.println("Usage: search <pattern> <file>");
+            System.exit(1);
+        }
+
+        String fileName = args[2];
+        String pattern = args[1];
+        try {
+            searchFile(fileName, pattern);
+        } catch (IOException e) {
```
Comment (AlexanderSandelius) [inline FileSearcher.java:29]: Specify Exception. Use FileNotFoundException.

File: testfile.txt [hunk 3]
```diff
@@ -0,0 +1,5 @@
+Hello how
```
Comment (AlexanderSandelius) [inline testfile.txt:1]: Should perhaps be a larger testfile.

**Review by AlexanderSandelius — CHANGES_REQUESTED** [review]

**Pull request merged**